src/main/java/net/coderbot/iris/shaderpack/IdMapParser.java

```diff
@@ -0,0 +1,176 @@
+package net.coderbot.iris.shaderpack;
+
+import com.google.common.collect.Maps;
+import net.coderbot.iris.mixin.AccessorRenderLayers;
+import net.minecraft.client.render.RenderLayer;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * A utility class for parsing entries in item.properties and block.properties files in shaderpacks
+ * This is not being used right now. Probably will delete it or reformat to be more efficient
+ */
+public class IdMapParser {
+
+    private Path shaderPath;
+    /**
+     * a map that contains the identifier of an item to the integer value parsed in block.properties
+     */
+    private Map<Identifier, Integer> blockPropertiesMap = Maps.newHashMap();
+    /**
+     * a map that contains the identifier of an item to the integer value parsed in item.properties
+     */
+    private Map<Identifier, Integer> itemPropertiesMap = Maps.newHashMap();
+    /**
+     * a map that contains render layers for blocks in block.properties
+     */
+    private Map<Identifier, RenderLayer> blockRenderLayerMap = Maps.newHashMap();
+
+    /**
+     * A map that contains the entity ids of an entity to the int value
+     */
+    private Map<Identifier, Integer> entityPropertiesMap = Maps.newHashMap();
+
+    IdMapParser(Path shaderPath){
+        this.shaderPath = shaderPath;
+        parseBlockProperties();
+        parseItemProperties();
+        parseEntityProperties();
+        this.blockRenderLayerMap.forEach((identifier, layer) -> AccessorRenderLayers.getBlockRenderLayers().put(Registry.BLOCK.get(identifier), layer));
+    }
+
+    /**
+     * Parses entries from entity.properties into the entityProperties map
+     */
+    private void parseEntityProperties(){
+        Properties properties = new Properties();
+        try {
+            properties.load(Files.newInputStream(shaderPath.resolve("entity.properties")));
+        } catch (IOException e) {
+            System.out.println("An entity.properties was not found in the current shaderpack");
```

Comment:
These probably shouldn't be printed to System.out, honestly. Maybe a debug message at most